core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -1336,7 +1341,7 @@ static int typeFromLegacyVersion(long version) {
     }
 
     public Type getType() {
-        return versionFromLegacyVersion(version) >= 3 ?
+        return versionFromLegacyVersion(version) >= MAX_STANDARD_VERSION ?
```

Comment:
This should be DIP2_VERSION, since MAX_STANDARD_VERSION may be raised to 4 at a later time.